app/controllers/admin/characters_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Admin::CharactersController < AdminController
+  before_action :authenticated_only
+  before_action :admin_only
+
+  def index
+    @characters = Character.validating
+  end
+
+  def update
+    Character.find(params[:id]).update!(character_params)
```

Comment:
`Character.find(params[:id])` move this to another method, e.g. `current_character`